src/slider/index.tsx

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect, useRef } from 'react';
+import clsx from 'clsx';
+
+import styles from './styles.css.js';
+import { getBaseProps } from '../internal/base-component';
+import { SliderProps } from './interfaces';
+import { applyDisplayName } from '../internal/utils/apply-display-name';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { fireNonCancelableEvent } from '../internal/events';
+import { useFormFieldContext } from '../internal/context/form-field-context';
+import { useUniqueId } from '../internal/hooks/use-unique-id';
+import Tooltip from '../internal/components/tooltip';
+import SliderLabels from './slider-labels.js';
+import { getPercent, getStepArray, findLowerAndHigherVals, THUMB_SIZE } from './utils.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+
+export { SliderProps };
+
+export default function Slider({
+  value,
+  min,
+  max,
+  onChange,
+  step,
+  disabled,
+  invalid,
+  ariaLabel,
+  referenceValues,
+  tickMarks,
+  hideFillLine,
+  valueFormatter,
+  ...rest
+}: SliderProps) {
+  const { __internalRootRef } = useBaseComponent('Slider');
+  const baseProps = getBaseProps(rest);
+
+  const rangeRef = useRef<HTMLDivElement>(null);
+  const handleRef = useRef<HTMLDivElement>(null);
+  const [showPopover, setShowPopover] = React.useState(false);
+  const [isActive, setIsActive] = React.useState(false);
+  const labelsId = useUniqueId('labels');
+  const hasValue = value !== undefined;
+  const formFieldContext = useFormFieldContext(rest);
+
+  const percent = hasValue && getPercent(Math.max(Math.min(value, max), min), min, max);
+
+  // at the min and max values, the thumb is not centered.
+  const tooltipPositionAdjust =
+    hasValue && (value <= min ? THUMB_SIZE - THUMB_SIZE / 2 : value >= max ? THUMB_SIZE + THUMB_SIZE / 2 : THUMB_SIZE);
+
+  useEffect(() => {
+    if (rangeRef.current) {
+      rangeRef.current.style.width = `${percent}%`;
```

Comment:
This probably doesn't need to be an effect. Can we apply this directly to the range element?